### PR TITLE
Replace requirement files with pyproject

### PR DESCRIPTION
**Pull request by dafeda**
## Pre review checklist

- [x] Read through the code changes carefully after finishing work
- [x] Make sure tests pass locally (after every commit!)
- [x] Prepare changes in small commits for more convenient review (optional)
- [x] PR title captures the intent of the changes, and is fitting for release notes.
- [ ] Updated documentation
- [ ] Ensured that unit tests are added for all new behavior (See 
    [Ground Rules](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md#ground-rules)),
    and changes to existing code have good test coverage.

## Pre merge checklist
- [x] Added appropriate release note label
- [x] Commit history is consistent and clean, in line with the [contribution guidelines](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md).

### CODE REVIEW

File: .github/workflows/style.yml [hunk 1]
```diff
@@ -32,11 +32,10 @@ jobs:
         cache-dependency-path: |
           setup.py
```
Comment (berland) [inline .github/workflows/style.yml:33]: this line was removed further up, it should not be removed here?

File: .github/workflows/style.yml [hunk 2]
```diff
@@ -32,11 +32,10 @@ jobs:
...
           pyproject.toml
-          style-requirements.txt
 
-    - name: Install dependencies
+    - name: Install ERT with style dependencies
```
Comment (berland) [inline .github/workflows/style.yml:36]: trailing whitespace

File: pyproject.toml [hunk 3]
```diff
@@ -12,7 +11,151 @@ requires = [
     "pybind11>=2.10.0",  # If this comes out of sync with the version installed by Conan please update the version in CMakeLists
     "grpcio-tools",
 ]
+build-backend = "setuptools.build_meta"
+
+[project]
```
Comment (berland) [inline pyproject.toml:16]: This section should perhaps be the first section in this file.

Comment (dafeda) [reply pyproject.toml:16]: https://setuptools.pypa.io/en/latest/userguide/pyproject_config.html

According to this, [build-system] is defined before [project]

File: .github/workflows/style.yml (hunk 2 above)
Comment (dafeda) [reply .github/workflows/style.yml]: Fixed

**Review by frode-aarstad — APPROVED** [review]
LGTM

**Pull request merged**